deploy/osps/default/osp-amzn2.yaml

```diff
@@ -111,96 +99,232 @@ spec:
+      {{- /* download cri-tools checksum */ }}
+      {{- /* the cri-tools checksum file has a filename prefix that breaks sha256sum so we need to drop it with sed */ }}
       cri_tools_sum=$(curl -Lf "$cri_tools_base_url/$cri_tools_filename.sha256" | sed 's/\*\///')
       cd "$opt_bin"
 
-      {{- /* verify cri-tools checksum */}}
+      {{- /* verify cri-tools checksum */ }}
       sha256sum -c <<<"$cri_tools_sum"
 
-      {{- /* unpack cri-tools and symlink to path so it's available to all users */}}
+      {{- /* unpack cri-tools and symlink to path so it's available to all users */ }}
       tar xvf "$cri_tools_filename"
       rm -f "$cri_tools_filename"
       ln -sf "$opt_bin/crictl" "$usr_local_bin"/crictl || echo "symbolic link is skipped"
       cd -
 
-      {{- /* kubelet */}}
+      {{- /* kubelet */ }}
       KUBE_VERSION="${KUBE_VERSION:-{{ .KubeVersion }}}"
       kube_dir="$opt_bin/kubernetes-$KUBE_VERSION"
       kube_base_url="https://storage.googleapis.com/kubernetes-release/release/$KUBE_VERSION/bin/linux/$arch"
       kube_sum_file="$kube_dir/sha256"
 
-      {{- /* create versioned kube dir */}}
+      {{- /* create versioned kube dir */ }}
       mkdir -p "$kube_dir"
       : >"$kube_sum_file"
 
       for bin in kubelet kubeadm kubectl; do
-          {{- /* download kube binary */}}
+          {{- /* download kube binary */ }}
           curl -Lfo "$kube_dir/$bin" "$kube_base_url/$bin"
           chmod +x "$kube_dir/$bin"
 
-          {{- /* download kube binary checksum */}}
+          {{- /* download kube binary checksum */ }}
           sum=$(curl -Lf "$kube_base_url/$bin.sha256")
 
-          {{- /* save kube binary checksum */}}
+          {{- /* save kube binary checksum */ }}
           echo "$sum  $kube_dir/$bin" >>"$kube_sum_file"
       done
 
-      {{- /* check kube binaries checksum */}}
+      {{- /* check kube binaries checksum */ }}
       sha256sum -c "$kube_sum_file"
 
       for bin in kubelet kubeadm kubectl; do
-          {{- /* link kube binaries from verioned dir to $opt_bin */}}
+          {{- /* link kube binaries from verioned dir to $opt_bin */ }}
           ln -sf "$kube_dir/$bin" "$opt_bin"/$bin
       done
 
-      if [[ ! -x /opt/bin/health-monitor.sh ]]; then
-          curl -Lfo /opt/bin/health-monitor.sh https://raw.githubusercontent.com/kubermatic/machine-controller/7967a0af2b75f29ad2ab227eeaa26ea7b0f2fbde/pkg/userdata/scripts/health-monitor.sh
-          chmod +x /opt/bin/health-monitor.sh
-      fi
+  files:
+    - path: /etc/environment
+      content:
+        inline:
+          data: |
+            {{- if .HTTPProxy }}
+            HTTP_PROXY={{ .HTTPProxy }}
+            http_proxy={{ .HTTPProxy }}
+            HTTPS_PROXY={{ .HTTPProxy }}
+            https_proxy={{ .HTTPProxy }}
+            NO_PROXY={{ .NoProxy }}
+            no_proxy={{ .NoProxy }}
+            {{- end }}
```

Comment:
Please revert this change since this would write an empty `/etc/environment` file if `HTTPProxy` is nil

Reply:
Is this a bad thing?

Reply:
OK, I see, previously it was appending to the file.

Reply:
OK, done, here and in other templates

Reply:
Thanks 😁